### PR TITLE
feat: add default install image

### PR DESCRIPTION
**Pull request by andrewrynhard**
We longer require an install image in the install config.

Signed-off-by: Andrew Rynhard <andrew@rynhard.io>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/2547)

### CODE REVIEW

File: pkg/machinery/config/types/v1alpha1/v1alpha1_provider.go [hunk 1]
```diff
@@ -927,6 +927,10 @@ func (t *TimeConfig) Servers() []string {
 
 // Image implements the config.Provider interface.
 func (i *InstallConfig) Image() string {
+	if i.InstallImage == "" {
```
Comment (andrewrynhard) [inline pkg/machinery/config/types/v1alpha1/v1alpha1_provider.go:930]: I couldn't find anywhere in the code that might block us from even allowing this (validation of the install image for example). If I am missing something, please let me know.

Comment (smira) [reply pkg/machinery/config/types/v1alpha1/v1alpha1_provider.go:930]: We could try keeping install image empty by default in talosctl cluster create (as a flag) to verify this

**Review by smira — APPROVED** [review]

File: pkg/machinery/config/types/v1alpha1/v1alpha1_provider.go (hunk 1 above)
Comment (andrewrynhard) [reply pkg/machinery/config/types/v1alpha1/v1alpha1_provider.go:930]: Ah, good idea.

**Review by Ulexus — APPROVED** [review]

**Review by smira — DISMISSED** [review]

**Review by smira — APPROVED** [review]

**Pull request merged**